From dcd9102079110feab34c71a8dc4cc484e29aaf64 Mon Sep 17 00:00:00 2001
From: Cupertino Miranda <cmiranda@synopsys.com>
Date: Wed, 30 Nov 2016 20:23:21 +0100
Subject: [PATCH 7/8] Fix for global undefined symbol in case of ARC_GOT32
 reloc. This was found with openssh and optreset configure test.

---
 bfd/elf32-arc.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/bfd/elf32-arc.c b/bfd/elf32-arc.c
index deec47dc9875..7e0093f5278b 100644
--- a/bfd/elf32-arc.c
+++ b/bfd/elf32-arc.c
@@ -1361,6 +1361,7 @@ elf_arc_relocate_section (bfd *		          output_bfd,
       asection *		    sec;
       struct elf_link_hash_entry *  h2;
       const char *                  msg;
+      bfd_boolean                   unresolved_reloc = FALSE;
 
       struct arc_relocation_data reloc_data =
       {
@@ -1448,6 +1449,14 @@ elf_arc_relocate_section (bfd *		          output_bfd,
 	}
       else
 	{
+	  bfd_boolean warned, ignored;
+	  bfd_vma relocation ATTRIBUTE_UNUSED;
+
+	  RELOC_FOR_GLOBAL_SYMBOL (info, input_bfd, input_section, rel,
+				   r_symndx, symtab_hdr, sym_hashes,
+				   h, sec, relocation,
+				   unresolved_reloc, warned, ignored);
+
 	  /* TODO: This code is repeated from below.  We should
 	     clean it and remove duplications.
 	     Sec is used check for discarded sections.
-- 
2.7.4

